purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -397,19 +397,25 @@ class Purchases internal constructor(
      * @param [upgradeInfo] The upgradeInfo you wish to upgrade from, containing the oldProductId and the optional
      * prorationMode. Amazon Appstore doesn't support changing products so upgradeInfo is ignored for Amazon purchases.
      * @param [listener] The PurchaseCallback that will be called when purchase completes.
+     * @param [isPersonalizedPrice] Set this parameter to true if the [SubscriptionOption] is a Google
+     * developer-determined offer available for purchase in the EU. Default is false. Ignored for Amazon.
+     * See https://developer.android.com/google/play/billing/integrate#personalized-price for more info.
      */
+    @JvmOverloads
     fun purchaseProduct(
         activity: Activity,
         storeProduct: StoreProduct,
         upgradeInfo: UpgradeInfo,
-        listener: ProductChangeCallback
+        listener: ProductChangeCallback,
+        isPersonalizedPrice: Boolean = false
```

Comment:
Can this parameter get moved to before the listener? 🤔  (Here and other places)

I _think_ the only way to do that would be to do something like...

```kotlin
fun purchaseProduct(
        activity: Activity,
        storeProduct: StoreProduct,
        upgradeInfo: UpgradeInfo,
        listener: ProductChangeCallback
    ) {
        this.purchaseProduct(activity, storeProduct, upgradeInfo, false, listener)
    }
```

```kotlin
 fun purchaseProduct(
        activity: Activity,
        storeProduct: StoreProduct,
        upgradeInfo: UpgradeInfo,
        isPersonalizedPrice: Boolean,
        listener: ProductChangeCallback
    ) {

}
```

I don't love the idea of having to add another method but I think as a developer it feels a lot better to have callbacks/lambdas as the last parameters 🙃

Reply:
If we move it to before the callbacks, then when we omit it (and use the default), [we have to used named parameters ](https://kotlinlang.org/docs/functions.html#default-arguments)for all the following parameters... i.e. 

```purchases.purchaseProduct(activity, storeProduct, upgradeInfo = upgradeInfo, listener = purchaseChangeCallback)```

```purchases.purchaseProduct(activity, storeProduct, upgradeInfo, purchaseChangeCallback)``` will have a type mismatch error :/

Reply:
oh, i see, your example fixes that 😓 yeah, i'm thinking that's the way to go too...